analyzers/tests/SonarAnalyzer.UnitTest/AnalysisContext/SonarAnalysisContextTest.ShouldAnalyzeGenerated.cs

```diff
@@ -48,48 +54,68 @@ public void ShouldAnalyzeGenerated_ResultIsCached()
-        sut.ShouldAnalyzeGenerated(compilation, options).Should().BeTrue();
-        sut.ShouldAnalyzeGenerated(compilation, options).Should().BeTrue();
+        sut.ShouldAnalyze(CSharpGeneratedCodeRecognizer.Instance, compilation.SyntaxTrees.Last(), compilation, options).Should().BeTrue();
+        sut.ShouldAnalyze(CSharpGeneratedCodeRecognizer.Instance, compilation.SyntaxTrees.Last(), compilation, options).Should().BeTrue();
+        sut.ShouldAnalyze(CSharpGeneratedCodeRecognizer.Instance, compilation.SyntaxTrees.Last(), compilation, options).Should().BeTrue();
 
         // GetText should be called every time ShouldAnalyzeGenerated is called...
         additionalText.Verify(x => x.GetText(It.IsAny<CancellationToken>()), Times.Exactly(3));
         sonarLintXml.ToStringCallCount.Should().Be(1); // ... but we should only try to read the file once
     }
 
-    [TestMethod]
-    public void ShouldAnalyzeGenerated_InvalidXmlInSonarLintFile_ReturnsFalse()
+    [DataTestMethod]
+    [DataRow(GeneratedFileName, false)]
+    [DataRow(OtherFileName, true)]
+    public void ShouldAnalyze_GeneratedFile_InvalidSonarLintXml(string fileName, bool expected)
     {
         var sonarLintXml = new DummySourceText("Not valid xml");
         var options = CreateOptions(sonarLintXml);
         var compilation = CreateDummyCompilation(AnalyzerLanguage.CSharp);
+        var tree = compilation.SyntaxTrees.Single(x => x.FilePath.Contains(fileName));
         var sut = CreateSut();
 
-        // 1. Read -> no error, false returned
-        sut.ShouldAnalyzeGenerated(compilation, options).Should().BeFalse();
+        // 1. Read -> no error
+        sut.ShouldAnalyze(CSharpGeneratedCodeRecognizer.Instance, tree, compilation, options).Should().Be(expected);
         sonarLintXml.ToStringCallCount.Should().Be(1); // should have attempted to read the file
 
         // 2. Read again to check that the load error doesn't prevent caching from working
-        sut.ShouldAnalyzeGenerated(compilation, options).Should().BeFalse();
+        sut.ShouldAnalyze(CSharpGeneratedCodeRecognizer.Instance, tree, compilation, options).Should().Be(expected);
         sonarLintXml.ToStringCallCount.Should().Be(1); // should not have attempted to read the file again
     }
 
-    [TestMethod]
-    public void ShouldAnalyzeGenerated_CorrectSettingUsed_VB()
+    [DataTestMethod]
+    [DataRow(GeneratedFileName)]
+    [DataRow(OtherFileName)]
+    public void ShouldAnalyze_GeneratedFile_AnalyzeGenerated_AnalyzeAllFiles(string fileName)
+    {
+        var sonarLintXml = CreateSonarLintXml(true);
+        var options = CreateOptions(sonarLintXml);
+        var compilation = CreateDummyCompilation(AnalyzerLanguage.CSharp);
+        var sut = CreateSut();
+
+        sut.ShouldAnalyze(CSharpGeneratedCodeRecognizer.Instance, compilation.SyntaxTrees.Single(x => x.FilePath.Contains(fileName)), compilation, options).Should().BeTrue();
```

Comment:
Could you extract a method with this: `compilation.SyntaxTrees.Single(x => x.FilePath.Contains(fileName))`? It's duplicated now in multiple tests.